pkg/agent/flannel/flannel.go

```diff
@@ -102,20 +103,34 @@ func LookupExtInterface(iface *net.Interface) (*backend.ExternalInterface, error
 		return nil, fmt.Errorf("failed to find IPv4 address for interface %s", iface.Name)
 	}
 
-	log.Infof("Using interface with name %s and address %s", iface.Name, ifaceAddr)
+	if netMode == (ipv4 + ipv6) {
+		ifacev6Addr, err = ip.GetInterfaceIP6Addr(iface)
+		if err != nil {
+			return nil, fmt.Errorf("failed to find IPv6 address for interface %s", iface.Name)
+		}
+
+		_, err := ip.GetDefaultV6GatewayInterface()
+
+		if err != nil {
+			return nil, fmt.Errorf("No ipv6 default route. Please add one")
```

Comment:
Is this truly required? What happens if we can't auto-detect the interface?

Users are probably going to want to be able to use IPv6 within the cluster on networks that don't have external ipv6 connectivity; I don't see any reason to make them add it just to pass this check.

Reply:
Yes, this is needed and if you don't provide it, things fail but you don't know why. That's why I added it. 

When using dual-stack, flannel creates two tunnels, one for ipv4 traffic and one for ipv6 traffic. The creation of both tunnels requires having a default route for both. It is common that the OS gives you a default route for ipv4 but it is not for ipv6. IIRC, if there is no route, the creation of the ipv6 tunnel fails silently

Reply:
Ok. Does that mean that we cannot support dual-stack clusters without full ipv6 connectivity? Can we support ipv6 between pods and services but not outside the cluster?

Reply:
Exactly, we cannot support dual-stack clusters without full ipv6 connectivity. An ipv4 vxlan tunnel does not allow to encapsulate ipv6 traffic. We need an ipv6 vxlan tunnel for that, that's why we need two tunnels. To create an ipv6 vxlan tunnel, we require full ipv6 connectivity across nodes

Reply:
I get why we need an IPv6 address on the node, but why does it need to have a default route? For IPv4 at least we allow using --node-ip and --flannel-iface to select an IP and interface to use if one cannot be auto-detected; we should be able to do the same for IPv6.

Reply:
I see your point and you are probably right. It's probably because of the autodetection of the interface but if we manually provide it, then it should not require that. Let me check flannel code again, I looked at this so long ago that I forgot the details

Reply:
You are 100% right. Thanks for noticing this. I'm removing that part of the code :)